<a href="https://colab.research.google.com/github/Anshikakhandelwall/DATA-ANALYTICS-ASSIGNMENTS-/blob/main/Decodelab1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# ============================================
# PROJECT 1 : DATA CLEANING & PREPARATION
# DecodeLabs Internship
# ============================================

# Step 1 : Import Libraries

import pandas as pd
import numpy as np


In [2]:
# --------------------------------------------
# Step 2 : Load Dataset
# --------------------------------------------

df = pd.read_excel("/content/Copy of Dataset for Data Analytics.xlsx")

print("Dataset Loaded Successfully")
print()

Dataset Loaded Successfully



In [3]:
# --------------------------------------------
# Step 3 : Basic Information
# --------------------------------------------

print("First 5 Rows")
print(df.head())

print("\nShape of Dataset")
print(df.shape)

print("\nColumn Names")
print(df.columns)

print("\nData Types")
print(df.dtypes)


First 5 Rows
     OrderID       Date CustomerID  Product  Quantity  UnitPrice  \
0  ORD200000 2023-01-04     C72649  Monitor         5     570.62   
1  ORD200001 2024-08-23     C75739    Phone         2     151.35   
2  ORD200002 2024-02-27     C81728   Tablet         5     550.68   
3  ORD200003 2023-10-15     C33540    Chair         1     273.19   
4  ORD200004 2025-05-08     C81840  Printer         4     626.01   

  ShippingAddress PaymentMethod OrderStatus TrackingNumber  ItemsInCart  \
0     928 Main St    Debit Card     Shipped    TRK37947903            7   
1     823 Main St        Online     Shipped    TRK91186779            3   
2     512 Main St   Credit Card   Cancelled    TRK42903982            8   
3     275 Main St    Debit Card    Returned    TRK62788070            5   
4     668 Main St        Online   Delivered    TRK29241424            8   

  CouponCode ReferralSource  TotalPrice  
0     SAVE10      Instagram     2853.10  
1     SAVE10       Referral      302.70  
2

In [4]:
# --------------------------------------------
# Step 4 : Check Missing Values
# --------------------------------------------

print("\nMissing Values in Each Column")

print(df.isnull().sum())

# Total Missing Values

print("\nTotal Missing Values")

print(df.isnull().sum().sum())


Missing Values in Each Column
OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

Total Missing Values
309


In [5]:
# --------------------------------------------
# Step 5 : Handle Missing Values
# --------------------------------------------

# Fill CouponCode missing values

df['CouponCode'] = df['CouponCode'].fillna("No Coupon")

print("\nMissing Values After Filling")

print(df.isnull().sum())



Missing Values After Filling
OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64


In [6]:
# --------------------------------------------
# Step 6 : Check Duplicate Rows
# --------------------------------------------

duplicates = df.duplicated().sum()

print("\nDuplicate Rows Found:")

print(duplicates)

# Remove Duplicates

df = df.drop_duplicates()

print("\nDuplicate Rows After Removal:")

print(df.duplicated().sum())



Duplicate Rows Found:
0

Duplicate Rows After Removal:
0


In [7]:
# --------------------------------------------
# Step 7 : Check Duplicate Order IDs
# --------------------------------------------

duplicate_orderid = df['OrderID'].duplicated().sum()

print("\nDuplicate Order IDs:")

print(duplicate_orderid)



Duplicate Order IDs:
0


In [8]:
# --------------------------------------------
# Step 8 : Correct Date Format
# --------------------------------------------

df['Date'] = pd.to_datetime(df['Date'], errors='coerce')

print("\nDate Column Converted Successfully")

print(df['Date'].head())


Date Column Converted Successfully
0   2023-01-04
1   2024-08-23
2   2024-02-27
3   2023-10-15
4   2025-05-08
Name: Date, dtype: datetime64[ns]


In [10]:
# --------------------------------------------
# Step 9 : Remove Extra Spaces
# --------------------------------------------

text_columns = df.select_dtypes(include='object').columns

for col in text_columns:

    df[col] = df[col].str.strip()

print("\nText Columns Standardized")



Text Columns Standardized


In [11]:
# --------------------------------------------
# Step 10 : Check Data Types Again
# --------------------------------------------

print("\nUpdated Data Types")

print(df.dtypes)




Updated Data Types
OrderID                    object
Date               datetime64[ns]
CustomerID                 object
Product                    object
Quantity                    int64
UnitPrice                 float64
ShippingAddress            object
PaymentMethod              object
OrderStatus                object
TrackingNumber             object
ItemsInCart                 int64
CouponCode                 object
ReferralSource             object
TotalPrice                float64
dtype: object


In [12]:
# --------------------------------------------
# Step 11 : Verify No Missing Values
# --------------------------------------------

print("\nFinal Missing Values")

print(df.isnull().sum())


Final Missing Values
OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64


In [ ]:
# --------------------------------------------
# Step 12 : Verify Duplicate Rows
# --------------------------------------------

print("\nFinal Duplicate Rows")

print(df.duplicated().sum())


In [13]:
# --------------------------------------------
# Step 13 : Verify Duplicate IDs
# --------------------------------------------

print("\nFinal Duplicate Order IDs")

print(df['OrderID'].duplicated().sum())


Final Duplicate Order IDs
0


In [14]:
# --------------------------------------------
# Step 14 : Dataset Summary
# --------------------------------------------

print("\nFinal Shape")

print(df.shape)

print("\nDataset Information")

print(df.info())


Final Shape
(1200, 14)

Dataset Information
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       1200 non-null   object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       


In [15]:
# --------------------------------------------
# Step 15 : Export Cleaned Dataset
# --------------------------------------------

df.to_csv("Cleaned_Data_Analytics.csv", index=False)

df.to_excel(
    "Cleaned_Data_Analytics.xlsx",
    index=False
)

print("\nCleaned Dataset Saved Successfully")

print("CSV File : Cleaned_Data_Analytics.csv")

print("Excel File : Cleaned_Data_Analytics.xlsx")


Cleaned Dataset Saved Successfully
CSV File : Cleaned_Data_Analytics.csv
Excel File : Cleaned_Data_Analytics.xlsx


OUTPUTS:

Total Rows               : 1200

Total Columns            : 14

Missing Values Found:
CouponCode : 309

Missing Values After Cleaning:
0

Duplicate Rows:
0

Duplicate Order IDs:
0

Incorrect Date Formats:
0

Status:
Dataset Successfully Cleaned and Prepared